lsp/roslyn_ls.lua

```diff
@@ -0,0 +1,195 @@
+            vim.notify(
+                "Razor is not supported.\nPlease use https://github.com/tris203/rzls.nvim",
+                vim.log.levels.WARN,
+                { title = "roslyn.nvim" }
+            )
+            return vim.NIL
+        end,
+    }
+end
+
+---@type vim.lsp.ClientConfig
+return {
+  name = "roslyn_ls",
+  offset_encoding = 'utf-8',
+  ---@param dispatchers vim.lsp.rpc.Dispatchers
+  cmd = function (dispatchers)
+    local pipe_name = "/tmp/422df9c8340645ba8966061884b388aa.sock"
+    vim.system({
+        "roslyn_ls", -- or provide the location of dll manually "dotnet", "Microsoft.CodeAnalysis.LanguageServer.dll",
+        "--logLevel",
+        "Trace",
+        "--extensionLogDirectory", -- this property is required by the server
+        fs.joinpath(uv.os_tmpdir(), 'roslyn_ls/logs'),
+        "--pipe",
+        pipe_name
+      })
+    uv.sleep(1500)
+    return vim.lsp.rpc.connect(pipe_name)(dispatchers)
+  end,
+  filetypes = { "cs" },
+  root_markers = { '.sln', '.csproj' },
+  handlers = roslyn_handlers(),
+  on_attach = function (client, bufnr)
+    -- enable inlay hints if LSP server supports it
+    if client.supports_method("textDocument/inlayHint") then
+      vim.lsp.inlay_hint.enable(true, { bufnr=bufnr })
+    end
+
+    -- enable semantic tokens highligting hints
+    if client.supports_method("textDocument/semanticTokens") then
+      client.server_capabilities.semanticTokensProvider = true
+    end
+
+    local bufname = vim.api.nvim_buf_get_name(bufnr)
+    -- don't try to find sln or csproj for files from libraries
+    -- outside of the project
+    if not bufname:match("^" .. fs.joinpath("/tmp/MetadataAsSource/")) then
+      -- try find solutions root first
+      local root_dir = nil
+      root_dir = fs.root(bufnr, function (name, path)
+        local match = name:match("%.sln$") ~= nil
+        if match then
+          local sln_file = fs.joinpath(path, name)
+          on_init_sln(client, sln_file)
+        end
+        return match
+      end)
+
+      if not root_dir then
+        -- try find solutions root first
+        root_dir = fs.root(bufnr, function (name, path)
+          local match = name:match("%.csproj$") ~= nil
+          if match then
+            local csproj_file = fs.joinpath(path, name)
+            on_init_project(client, {csproj_file})
+          end
+          return match
+        end)
+      end
+
+      assert(root_dir, "no solution and no csproj found")
+
+      client.root_dir = root_dir
+    end
+  end,
+  capabilities = {
+      -- HACK: Enable filewatching to later just not watch any files
+      -- This is to not make the server watch files and make everything super slow in certain situations
+      workspace = {
+          didChangeWatchedFiles = {
+              dynamicRegistration = true,
+              -- enable file watcher capabilities for lsp clients
+              relativePatternSupport = true,
+          },
+      },
```

Comment:
This should also just be removed IMO. I have some config for turning it off/on in the plugin because filewatching can be very slow. However, I don't think `nvim-lspconfig` should have any of this